icu4c/source/common/caniter.cpp

```diff
@@ -490,6 +493,15 @@ Hashtable *CanonicalIterator::getEquivalents2(Hashtable *fillinResult, const cha
 
                 ne = remainder.nextElement(el);
             }
+            // ICU-22642 To avoid security attack by an useless string which
+            // have too many permutations to hang the function, limit the
+            // return size and return error with U_UNSUPPORTED_ERROR if exceed
+            // the limit.
+            constexpr int32_t kResultLimit = 4096;
+            if (fillinResult->count() > kResultLimit) {
+                status = U_UNSUPPORTED_ERROR;
```

Comment:
Don't we normally return `U_ILLEGAL_ARGUMENT_ERROR` in cases like this, or is there somthing about this kind of input that would make that status inappropriate here?

Reply:
we already use when fix 
https://github.com/unicode-org/icu/commit/757d2cd90a434b35596d04fc89421dd0add33952